zstd_stream.go

```diff
@@ -267,24 +254,18 @@ func (w *Writer) Close() error {
 
 	ret := 1 // So we loop at least once
 	for ret > 0 {
-		var srcPtr *byte // Do not point anywhere, if src is empty
-		if len(w.srcBuffer) > 0 {
-			srcPtr = &w.srcBuffer[0]
-		}
-
 		C.ZSTD_compressStream2_finish(
 			w.resultBuffer,
 			w.ctx,
 			unsafe.Pointer(&w.dstBuffer[0]),
 			C.size_t(len(w.dstBuffer)),
-			unsafe.Pointer(srcPtr),
-			C.size_t(len(w.srcBuffer)),
+			unsafe.Pointer(uintptr(0)),
```

Comment:
same here? If there's never anything left to flush can we simplify this and not call compressStream at all and just return any known error / free any resources needed?

Reply:
Nope, same reason as above.